api/lib/opentelemetry/trace/span.rb

```diff
@@ -6,42 +6,142 @@
+      # meanings.
+      #
+      # @param [String, Symbol] key
+      # @param [String, Symbol, Boolean, Numeric] value
+      #
+      # @return [self] returns itself
       def set_attribute(key, value)
         raise ArgumentError unless valid_key?(key)
         raise ArgumentError unless valid_value?(value)
+
+        self
       end
       alias []= set_attribute
 
+      # Add an Event to Span
+      #
+      # Note that the OpenTelemetry project
+      # {https://github.com/open-telemetry/opentelemetry-specification/blob/master/semantic-conventions.md
+      # documents} certain "standard event names and keys" which have
+      # prescribed semantic meanings.
+      #
+      # @param [String] name name of the event
+      # @param [Hash<String, Object>] attrs One or more key:value pairs, where
+      #   the keys must be strings and the values may be string, boolean or
+      #   numeric type.
+      #
+      # @return [self] returns itself
       def add_event(name, **attrs)
         raise ArgumentError if name.nil?
+
+        self
       end
 
+      # Adds a link to another Span from this Span. Linked Span can be from the
+      # same or different trace. See Links description.
+      #
+      # @param [SpanContext, Link] span_context_or_link SpanContext of the Span
+      #   to link to or a Link which allows allows instrumentation to supply a
+      #   lazily calculated Link.
+      # @param [Hash<String, Object>] attrs Map of attributes associated with
+      #   this link. Attributes are key:value pairs where key is a string and
+      #   value is one of string, boolean or numeric type.
+      #
+      # @return [self] returns itself
       def add_link(span_context_or_link, **attrs)
         raise ArgumentError if span_context_or_link.nil?
         raise ArgumentError unless span_context_or_link.instance_of?(SpanContext) || attrs.empty?
+
+        self
       end
 
+      # Sets the Status to the Span
+      #
+      # If used, this will override the default Span status. Default is OK.
+      #
+      # Only the value of the last call will be recorded, and implementations
+      # are free to ignore previous calls.
+      #
+      # @param [Object] status - TODO: Change Object to actual Status enum when
+      #   introduced to the api
+      #
+      # @return [void]
       def status=(status)
         raise ArgumentError if status.nil?
       end
 
+      # Updates the Span name
+      #
+      # Upon this update, any sampling behavior based on Span name will depend
+      # on the implementation.
+      #
+      # @param [String] new_name The new operation name, which supersedes
+      #   whatever was passed in when the Span was started
+      #
+      # @return [void]
       def name=(new_name)
         raise ArgumentError if new_name.nil?
       end
 
-      def finish; end
+      # Finishes the Span
+      #
+      # This call will take the current timestamp to set as Span's end time.
```

Comment:
This sentence is ambiguous (in the spec too). It **will** be correct once https://github.com/open-telemetry/rfcs/pull/8 propagates to the spec, but for now it suggests the method takes the current timestamp as an argument when in fact it has no parameters.

Reply:
removed this sentence for now